### ISSUE TITLE
[AVM Module Issue]: InvalidKeyRotationPolicy

### ISSUE THREAD

**Issue by timdhaeyer** [Azure/bicep-registry-modules#2760]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/key-vault/vault

### (Optional) Module Version

0.6.2

### Description

When deploying a Keyvault resource together with a single key I get the following error.

> "code":"InvalidKeyRotationPolicy","message":"The given value for 'lifetimeActions' is null."

This is the key I'm passing through. No Rotation policy is defined, so there must be something wrong with the defaults.
```
var kv02Keys = [
  //Customer Managed Key (CMK) for data encryption
  {
    name: cmkName
    attributeEnabled: true
    keyOps: [
      'wrapKey'
      'unwrapKey'
    ]
    keySize: 2048
    kty: 'RSA'
    tags: {
      'hidden-title': 'Key for data encryption in rest (storage, datalake)'
    }
  }
]
```

**Comment by AlexanderSehr** [comment]
Hey @fblix, please take a look an triage the issue when you have a moment 💪

**Comment by timdhaeyer** [comment]
Nobody? :-)

**Comment by AlexanderSehr** [comment]
Hey @timdhaeyer, certainly @fblix. I'll reach out directly in case he overlooked this issue :)

**Comment by fblix** [comment]
Hey @timdhaeyer sorry I absolutely overlooked this one, thanks @AlexanderSehr for pinging me internally.

Will try to reproduce this and come back to you. In the meantime, if anyone else wants to chime in, feel free to do so.

**Comment by fblix** [comment]
@marcingastol have a look here

**Comment by marcingastol** [comment]
Hi @timdhaeyer, I’ve begun working on the issue you reported and will start from replicating it on my end. I’ll keep you updated on my progress.

**Comment by timdhaeyer** [comment]
@marcingastol any updates on this?

**Comment by margas-expinit** [comment]
Hi @timdhaeyer and @fblix I'm still working on this. 

Successfully replicated user issue (AKV version module 0.7.1.):
- can't deploy AKV with key if rotationPolicy is empty (The given value for 'lifetimeActions' is null)
- can't deploy AKV with key if lifetimeActions is empty (The given value for 'lifetimeActions' is null)

Tested deployment of AKV with key by native AKV bicep module and ARM template without rotationPolicy and lifetimeActions - with success. Issue definitely is in AVM module.
Tested deployment of AKV with key on 0.6.0 AVM module version - the same problem as on 0.7.1
Tested deployment of AKV with key on 0.5.0 AVM module version - no issue with deployment

Tested:
1. Azure Bicep module:
1.1. If rotationPolicy is added, also lifetimeActions is must
1.2. If rotationPolicy is added with lifetimeAction too, action + trigger are must

**Comment by marcingastol** [comment]
Hi @timdhaeyer and @fblix - I'm still working on this. 

Successfully replicated user issue (AKV version module 0.7.1.):
- can't deploy AKV with key if rotationPolicy is empty (The given value for 'lifetimeActions' is null)
- can't deploy AKV with key if lifetimeActions is empty (The given value for 'lifetimeActions' is null)

Tested deployment of AKV with key by native AKV bicep module and ARM template without rotationPolicy and lifetimeActions - with success. Issue definitely is in AVM module.
Tested deployment of AKV with key on 0.6.0 AVM module version - the same problem as on 0.7.1
Tested deployment of AKV with key on 0.5.0 AVM module version - no issue with deployment

Tested:
1. Azure Bicep module:
1.1. If rotationPolicy is added, also lifetimeActions is must
1.2. If rotationPolicy is added with lifetimeAction too, action + trigger are must

**Comment by fblix** [comment]
fixed through #3381
